jdaviz/core/helpers.py

```diff
@@ -314,32 +315,43 @@ def show(self, loc="inline", title=None):
         If "sidecar" is requested in the "classic" Jupyter notebook, the app will appear inline,
         as only JupyterLab has a mechanism to have multiple tabs.
         """
-        try:
-            if loc == "inline":
+        # Check if the user is running Jdaviz in the correct environments.
+        # If not, provide a friendly msg to guide them!
+        if get_ipython().__class__.__name__ != 'ZMQInteractiveShell':
+            boilerplate_e = \
+                RuntimeError("\nYou are currently running Jdaviz from an unsupported "
+                             "shell. Jdaviz is intended to be run within a Jupyter "
+                             "notebook, or directly from the command line.\n\n"
+                             "To run from Jupyter, call <your viz>.show() from a notebook cell.\n"
+                             "To see how to run from the command line, run: "
+                             "'jdaviz --help' outside of Python.\n\n"
+                             "To learn more, see our documentation at: "
+                             "https://jdaviz.readthedocs.io\n\n"
+                             "Thanks for trying out Jdaviz! :)")
+            raise boilerplate_e
```

Comment:
Smiley face sometimes has the opposite effect (reverse psychology?), so let's make the message less personal.
```suggestion
            raise RuntimeError("\nYou are currently running Jdaviz from an unsupported "
                               f"shell ({cur_shell_name}). Jdaviz is intended to be run within a Jupyter "
                               "notebook, or directly from the command line.\n\n"
                               "To run from Jupyter, call <your viz>.show() from a notebook cell.\n"
                               "To see how to run from the command line, run: "
                               "'jdaviz --help' outside of Python.\n\n"
                               "To learn more, see our documentation at: "
                               "https://jdaviz.readthedocs.io")
```

Reply:
I think this is what Kyle suggested above as well (the raising directly part).